src/ansys/additive/core/parametric_study/display/porosity_eval_plot.py

```diff
@@ -0,0 +1,325 @@
+# Copyright (C) 2023 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+# of the Software, and to permit persons to whom the Software is furnished to do
+# so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from __future__ import annotations
+
+import math
+import os
+
+import numpy as np
+import pandas as pd
+import panel as pn
+import plotly.graph_objects as go
+
+from ansys.additive.core import SimulationStatus, SimulationType
+from ansys.additive.core.misc import short_uuid
+from ansys.additive.core.parametric_study import ColumnNames, ParametricStudy
+
+from ._common_controls import _common_controls
+
+# global variables
+_range_slider = None
+
+pn.extension("plotly")
+
+
+def porosity_eval_plot(ps: ParametricStudy):
+    """Generate a heat map plot of porosity results to determine parametric
+    regions with desirable relative density statistics.
+
+    Parameters
+    ----------
+    ps : :class:`ParametricStudy <from ansys.additive.core.parametric_study.ParametricStudy>`
+        Parametric study to plot.
+
+    Returns
+    -------
+    :class: `panel.Row <panel.Row>`
+        Interactive plot.
+    """
```

Comment:
```suggestion
    """ # noqa
```
This will prevent the long line above from being flagged by the pre-commit checks.